acceptancetests/assess_lxd_space.py

```diff
@@ -0,0 +1,179 @@
+   This is an validation of Bug 1685782:
+   https://bugs.launchpad.net/juju/+bug/1685782
+   The test procedures are based on comment #29.
+"""
+
+from __future__ import print_function
+
+import os
+import sys
+import json
+import logging
+import argparse
+
+from jujucharm import local_charm_path
+from deploy_stack import BootstrapManager
+
+from utility import (
+    JujuAssertionError,
+    add_basic_testing_arguments,
+    configure_logging,
+    )
+
+__metaclass__ = type
+log = logging.getLogger("assess_lxd_space")
+
+
+def assert_initial_spaces(client):
+    """Initial spaces status after bootstrap should show as:
+       no spaces to display"""
+    # Run juju spaces --format json
+    # Bug 1704105: https://bugs.launchpad.net/juju/+bug/1704105
+    # merge_stderr need to be set to True for the initial assert
+    # because spaces list is empty at this point.
+    raw_output = client.get_juju_output(
+        'spaces', '--format', 'json', include_e=False, merge_stderr=True)
+    expected_spaces = 'no spaces to display'
+    if expected_spaces not in raw_output:
+        raise JujuAssertionError(
+            'Incorrect initial spaces status. Found: {}\nExpected: {}'.format(
+            raw_output, expected_spaces))
+    else:
+        log.info('Initial spaces status is: {}.'.format(expected_spaces))
+
+
+def get_subnets(client):
+    # Run juju subnets --format json
+    raw_output = client.get_juju_output(
+        'subnets', '--format', 'json', include_e=False)
+    try:
+        subnets_output = json.loads(raw_output)
+    except ValueError as e:
+        log.error('Invalid output from juju subnets:\n{}'.format(raw_output))
+        raise e
+    subnets_dict = subnets_output['subnets']
+    subnets_cidr_list = subnets_output['subnets'].keys()
+    return (subnets_dict, subnets_cidr_list)
+
+
+def assert_initial_subnets(client):
+    """There should be at least 1 subnet available once bootstrap is completed.
+       By default 4 subnets should be shown under an AWS controller."""
+    subnets_dict = get_subnets(client)[0]
+    if not subnets_dict:
+        raise JujuAssertionError('No subnet can be Found.')
+    else:
+        log.info('{} subnet(s) have been found.'.format(str(len(subnets_dict))))
+
+
+def add_space_with_existing_subnet(client, space_name):
+    # Run juju add-space <space_name> <subnet_cidr>
+    subnets_cidr_list = get_subnets(client)[1]
+    subnet_cidr = subnets_cidr_list[0]
+    # Bug 1704105, merge_stderr=True is required.
+    client.get_juju_output(
+        'add-space', space_name, subnet_cidr, include_e=False, merge_stderr=True)
+    return (space_name, subnet_cidr)
+
+
+def assert_added_space(client):
+    """Validate if configurations in new added space are as expected.
+       Five checkpoints:
+       1. After new space is added, juju spaces --format json should give output
+          in json format, in comparison to no space exists.
```

Comment:
Please file a bug around this and --format yaml. It shouldn't hold this landing.